File: Atividades Problemas de Programação - sockets 1/Exercicio 9/Servidor.java
import java.net.*;
import java.util.Arrays;
import java.io.*;

class Conexao {

	private Socket socketCliente;

	Conexao(Socket aSocketCliente) throws IOException {
		this.socketCliente = aSocketCliente;
	}

	public void iniciar() {
		PrintWriter saida = null;
		BufferedReader entrada = null;

		InetAddress endCliente = this.socketCliente.getInetAddress();

		int valor = 0;
		int naipe = 0;
		String nome = null;
		String[] mensagem = null;

		try {
			saida = new PrintWriter(this.socketCliente.getOutputStream(), true);
			entrada = new BufferedReader(new InputStreamReader(this.socketCliente.getInputStream()));

			mensagem = entrada.readLine().split(",");
			valor = Integer.parseInt(mensagem[0]);
			naipe = Integer.parseInt(mensagem[1]);

			switch (valor) {
			case 1:
				nome = "As de ";
				break;
			case 2:
				nome = "Dois de ";
				break;
			case 3:
				nome = "Tres de ";
				break;
			case 4:
				nome = "Quatro de ";
				break;
			case 5:
				nome = "Cinco de ";
				break;
			case 6:
				nome = "Seis de ";
				break;
			case 7:
				nome = "Sete de ";
				break;
			case 8:
				nome = "Oito de ";
				break;
			case 9:
				nome = "Nove de ";
				break;
			case 10:
				nome = "Dez de ";
				break;
			case 11:
				nome = "Valete de ";
				break;
			case 12:
				nome = "Dama de ";
				break;
			case 13:
				nome = "Rei de ";
				break;
			}

			switch (naipe) {
			case 1:
				nome = nome + "ouros";
				break;
			case 2:
				nome = nome + "paus";
				break;
			case 3:
				nome = nome + "copas";
				break;
			case 4:
				nome = nome + "espadas";
				break;
			}

			saida.println(nome);

			System.out.println(
					"Cliente " + endCliente.getHostAddress() + " atendido com a mensagem " + Arrays.toString(mensagem));

			socketCliente.close();
			saida.close();
			entrada.close();

		} catch (IOException e) {
			System.out.println("Erro E/S " + e);
		}
	}
}

public class Servidor {
	public static void main(String[] args) throws IOException {

		final int portaDefault = 8080;
		int porta;

		Socket socketCliente = null;
		ServerSocket socketServidor = null;

		if ((args.length == 1))
			porta = Integer.parseInt(args[0]);
		else
			porta = portaDefault;

		while (true) {
			try {
				socketServidor = new ServerSocket(porta);
				break;
			} catch (IOException e) {
				porta++;
			}
		}

		System.out.println("\nServidor ativado. " + "Aguardando Cliente na porta " + porta + "...\n");

		while (true) {

			socketCliente = null;
			try {
				socketCliente = socketServidor.accept();
			} catch (IOException e) {
				System.err.println("Erro de E/S " + e);
				socketServidor.close();
				System.exit(1);
			}

			new Conexao(socketCliente).iniciar();
		}
	}
}

File: Atividades Problemas de Programação - rpc_rmi/RPC/Exercicio 9/servidor.py
from xmlrpc import server
from xmlrpc.server import SimpleXMLRPCServer


def carta(valor, naipe):
    if valor < 1 or valor > 13:
        return('Valor incorreto, somente de 1 a 13')
    elif naipe < 1 or naipe > 4:
        return('Naipe incorreto, somente de 1 a 4')
    else:
        valores = {1: 'Ás', 2: 'Dois', 3: 'Três', 4: 'Quatro',
                   5: 'Cinco', 6: 'Seis', 7: 'Sete', 8: 'Oito', 9: 'Nove',
                   10: 'Dez', 11: 'Valete', 12: 'Dama', 13: 'Rei'}
        naipes = {1: 'Ouros', 2: 'Paus', 3: 'Copas', 4: 'Espadas'}

        nome = valores[valor]+' de '+naipes[naipe]
        return(nome)


server = SimpleXMLRPCServer(("localhost", 8080))
print("Listening on port 8080...")
server.register_function(carta, 'carta')

server.serve_forever()
